File: src/mocks/handlers/audit.ts
import { http, HttpResponse } from 'msw'
import { API_BASE_PATH } from '@/config/app'
import { paginate } from '../lib/paginate'
import { parseIntParam } from '../lib/parse'
import { mockOperationLogs, mockCallLogs } from '../data'

export const auditHandlers = [
  // ========== 审计日志 ==========
  http.get(`${API_BASE_PATH}/audit/operations`, ({ request }) => {
    const url = new URL(request.url)
    const action = url.searchParams.get('action')
    const page = parseIntParam(url.searchParams.get('page'), 1)
    const pageSize = parseIntParam(url.searchParams.get('pageSize'), 20)
    const items = action ? mockOperationLogs.filter((l) => l.action === action) : mockOperationLogs
    return HttpResponse.json(paginate(items, page, pageSize))
  }),
  http.get(`${API_BASE_PATH}/audit/calls`, ({ request }) => {
    const url = new URL(request.url)
    const model = url.searchParams.get('model')
    const status = url.searchParams.get('status')
    const page = parseIntParam(url.searchParams.get('page'), 1)
    const pageSize = parseIntParam(url.searchParams.get('pageSize'), 20)
    let items = mockCallLogs
    if (model) items = items.filter((l) => l.model === model)
    if (status) items = items.filter((l) => l.status === status)
    return HttpResponse.json(paginate(items, page, pageSize))
  }),
]

File: apps/backend/internal/http/handler/audit.go
package handler

import (
	"net/http"

	"github.com/go-chi/chi/v5"
	"github.com/tokenjoy/backend/internal/config"
	domainaudit "github.com/tokenjoy/backend/internal/domain/audit"
	"github.com/tokenjoy/backend/internal/domain/session"
	"github.com/tokenjoy/backend/internal/domain/types"
	"github.com/tokenjoy/backend/internal/http/httputil"
	httpmiddleware "github.com/tokenjoy/backend/internal/http/middleware"
	"github.com/tokenjoy/backend/internal/permission"
	pkg "github.com/tokenjoy/backend/internal/pkg"
)

type AuditHandler struct {
	sessionHandlerBase
	service domainaudit.Service
}

func NewAuditHandler(cfg config.Config, service domainaudit.Service, sessionSvc session.Service) *AuditHandler {
	return &AuditHandler{
		sessionHandlerBase: newSessionHandlerBase(cfg, sessionSvc),
		service:            service,
	}
}

func (h *AuditHandler) SettingsGet(w http.ResponseWriter, r *http.Request) {
	httputil.WriteOK(w, h.service.GetSettings())
}

func (h *AuditHandler) SettingsUpdate(w http.ResponseWriter, r *http.Request) {
	var body types.AuditSettings
	if err := httputil.DecodeJSON(r, &body); err != nil {
		httputil.WriteError(w, err)
		return
	}
	settings, err := h.service.UpdateSettings(body)
	httputil.WriteJSON(w, http.StatusOK, settings, err)
}

func (h *AuditHandler) OperationsList(w http.ResponseWriter, r *http.Request) {
	query := r.URL.Query()
	params := types.AuditOperationsQueryParams{
		Page:       pkg.ParseIntParam(query.Get("page"), 1),
		PageSize:   pkg.ParseIntParam(query.Get("pageSize"), 20),
		Action:     query.Get("action"),
		OperatorID: query.Get("operatorId"),
		Keyword:    query.Get("keyword"),
		From:       query.Get("from"),
		To:         query.Get("to"),
	}
	httputil.WriteOK(w, h.service.ListOperations(params))
}

func (h *AuditHandler) CallsList(w http.ResponseWriter, r *http.Request) {
	query := r.URL.Query()
	params := types.AuditCallsQueryParams{
		Page:     pkg.ParseIntParam(query.Get("page"), 1),
		PageSize: pkg.ParseIntParam(query.Get("pageSize"), 20),
		Model:    query.Get("model"),
		Status:   query.Get("status"),
		CallerID: query.Get("callerId"),
		Keyword:  query.Get("keyword"),
		From:     query.Get("from"),
		To:       query.Get("to"),
	}
	httputil.WriteOK(w, h.service.ListCalls(params))
}

func (h *AuditHandler) RegisterRoutes(r chi.Router) {
	read := httpmiddleware.PublicOrReadRoutes(h.cfg, r, h.sessionSvc, permission.AuditRead)
	read.Get("/settings", h.SettingsGet)
	read.Get("/operations", h.OperationsList)
	read.Get("/calls", h.CallsList)

	httpmiddleware.WriteRoutes(r, h.cfg, h.sessionSvc).
		With(httpmiddleware.RequireAnyPermission(permission.AuditRead)).
		Put("/settings", h.SettingsUpdate)
}
